internal/controller/organization/service_proxy_test.go

```diff
@@ -99,7 +99,27 @@ var _ = Describe("Organization ServiceProxyReconciler", Ordered, func() {
 		})
 	})
 
-	When("organization is annotated with the oauth2proxy preview annotation", func() {
+	When("organization is annotated with the oauth2proxy preview annotation", Ordered, func() {
+		It("should create default organization with oidc", func() {
+			By("creating greenhouse organization with OIDC config")
+			defaultOrg := setup.CreateOrganization(test.Ctx, "greenhouse", func(org *greenhousev1alpha1.Organization) {
+				org.Spec.MappedOrgAdminIDPGroup = "SOME_IDP_GROUP"
+			})
+			test.EventuallyCreated(test.Ctx, test.K8sClient, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: defaultOrg.Name}})
+			By("creating a secret with dummy oidc config for default organization")
+			secret := setup.CreateSecret(test.Ctx, "greenhouse-oidc-config", test.WithSecretNamespace(defaultOrg.Name), test.WithSecretData(map[string][]byte{"clientID": []byte("dummy"), "clientSecret": []byte("top-secret")}))
+
+			By("Updating the default organization with oidc config")
+			defaultOrg = setup.UpdateOrganization(test.Ctx, defaultOrg.Name, test.WithOIDCConfig("some-issuer.tld", secret.Name, "clientID", "clientSecret"))
```

Comment:
Good one, this would be another one to gradually reduce the repetitive code for updating the Greenhouse resources across our tests 👍